libraries/ESP8266WebServer/src/Parsing-impl.h

```diff
@@ -62,215 +62,182 @@ static bool readBytesWithTimeout(typename ServerType::ClientType& client, size_t
+    DEBUG_OUTPUT.print(" search: ");
+    DEBUG_OUTPUT.println(searchStr);
 #endif
 
-  //attach handler
-  RequestHandlerType* handler;
-  for (handler = _firstHandler; handler; handler = handler->next()) {
-    if (handler->canHandle(_currentMethod, _currentUri))
-      break;
-  }
-  _currentHandler = handler;
+    //attach handler
+    RequestHandlerType* handler;
+    for (handler = _firstHandler; handler; handler = handler->next()) {
+        if (handler->canHandle(_currentMethod, _currentUri))
+            break;
+    }
+    _currentHandler = handler;
 
-  String formData;
-  // below is needed only when POST type request
-  if (method == HTTP_POST || method == HTTP_PUT || method == HTTP_PATCH || method == HTTP_DELETE){
-    String boundaryStr;
     String headerName;
     String headerValue;
+    String boundaryStr;
     bool isForm = false;
     bool isEncoded = false;
     uint32_t contentLength = 0;
-    //parse headers
-    while(1){
-      req = client.readStringUntil('\r');
-      client.readStringUntil('\n');
-      if (req == "") break;//no moar headers
-      int headerDiv = req.indexOf(':');
-      if (headerDiv == -1){
-        break;
-      }
-      headerName = req.substring(0, headerDiv);
-      headerValue = req.substring(headerDiv + 1);
-      headerValue.trim();
-       _collectHeader(headerName.c_str(),headerValue.c_str());
-
-      #ifdef DEBUG_ESP_HTTP_SERVER
-      DEBUG_OUTPUT.print("headerName: ");
-      DEBUG_OUTPUT.println(headerName);
-      DEBUG_OUTPUT.print("headerValue: ");
-      DEBUG_OUTPUT.println(headerValue);
-      #endif
-
-      if (headerName.equalsIgnoreCase(FPSTR(Content_Type))){
-        using namespace mime;
-        if (headerValue.startsWith(FPSTR(mimeTable[txt].mimeType))){
-          isForm = false;
-        } else if (headerValue.startsWith(F("application/x-www-form-urlencoded"))){
-          isForm = false;
-          isEncoded = true;
-        } else if (headerValue.startsWith(F("multipart/"))){
-          boundaryStr = headerValue.substring(headerValue.indexOf('=') + 1);
-          boundaryStr.replace("\"","");
-          isForm = true;
-        }
-      } else if (headerName.equalsIgnoreCase(F("Content-Length"))){
-        contentLength = headerValue.toInt();
-      } else if (headerName.equalsIgnoreCase(F("Host"))){
-        _hostHeader = headerValue;
-      }
-    }
+    while (true) {
+        req = client.readStringUntil('\r');
+        client.readStringUntil('\n');
 
-    String plainBuf;
-    if (   !isForm
-        && // read content into plainBuf
-           (   !readBytesWithTimeout<ServerType>(client, contentLength, plainBuf, HTTP_MAX_POST_WAIT)
-            || (plainBuf.length() < contentLength)
-           )
-       )
-    {
-        return false;
-    }
+        int headerDiv = req.indexOf(':');
+        if (headerDiv < 0) {
+            break;
+        }
+        headerName = req.substring(0, headerDiv);
+        headerValue = req.substring(headerDiv + 2);
```

Comment:
I think you're assuming there is always a space after the ":" with this change.  I'm not sure I see where in the RFC that whitespace is guaranteed, tough.  Am I missing something?

Reply:
reverted. I think that was an accidental CTRL-A too many